smrt/rtsolver/multifresnel_thermalemission.py

```diff
@@ -123,12 +123,24 @@ def solve(self, snowpack, emmodels, sensor, atmosphere=None):
             prune_deep_snowpack=self.prune_deep_snowpack,
         )
 
+        if min_optical_depth < 5 and snowpack.substrate is None :
+            smrt_warn(
+                "Multifresnel has detected that the snowpack is optically shallow (tau=%g) and no substrate has been set, meaning that the space "
+                "under the snowpack is empty, the last layer bottom interface might be missing and"
+                " the snowpack is shallow enough to affect the signal measured at the surface."
+                "This is usually not wanted and can produce wrong results. Either increase the thickness of the snowpack or set a substrate."
+                " If wanted, add a transparent substrate to supress this warning" % min_optical_depth
+            )
+
         Tbv, Tbh = compute_emerging_radiation(M)
 
         #  describe the results list of (dimension name, dimension array of value)
         coords = [("theta", sensor.theta_deg), ("polarization", ["V", "H"])]
 
         # store other diagnostic information
-        other_data = prepare_kskaeps_profile_information(snowpack, emmodels, effective_permittivity, mu=mu)
+        if snowpack.substrate is None:
+            other_data = prepare_kskaeps_profile_information(snowpack, emmodels, effective_permittivity, mu=mu)
+        else:
+            other_data = prepare_kskaeps_profile_information(snowpack, emmodels, effective_permittivity[:-1], mu=mu)
```

Comment:
remove the if
and change effective_permittivity => effective_permittivity[0:snowpack.nlayers] for a more compact alternative. Along with the change "temperature = np.append(temperature, snowpack.substrate.temperature)" this is should be an independent commit (not including the the calculation of the optical depth, which will be refactored later)